src/provider.c

```diff
@@ -216,17 +218,17 @@ static int provider_get_params(void *provctx, OSSL_PARAM params[])
             if (params->data_type != OSSL_PARAM_UTF8_PTR)
                 return 0;
 
-	    params->data = ctx->provider_name;
+            *((char **)params->data) = ctx->provider_name;
             params->return_size = strlen(ctx->provider_name) + 1;
-	    continue;
+            continue;
         }
         if (strcmp(params->key, OSSL_PROV_PARAM_VERSION) == 0) {
             if (params->data_type != OSSL_PARAM_UTF8_PTR)
                 return 0;
 
-	    params->data = VERSION;
+            *((char **)params->data) = VERSION;
```

Comment:
Why would one want to cast to char ** and then dereference here ?

Reply:
Found the explaination in OSSL_PARAM(3)
```
The difference between this [OSSL_PARAM_UTF8_PTR] and OSSL_PARAM_UTF8_STRING is that data doesn't point directly at the data, but to a pointer that points to the data.
```

Reply:
Because `params->data` for `OSSL_PARAM_UTF8_PTR` will point to a buffer where the pointer to the string must be stored.

Comparing with get_utf8_ptr: https://github.com/openssl/openssl/blob/master/crypto/params.c#L860-L874

So it's either `*(params->data) = (const void *) VERSION` or casting `params->data` to the pointer type for `VERSION`.

I would prefer the casting `params->data` because that way the compiler will raise a warning if in the future `VERSION` gets redefined and it's type is not a `const char*` anymore.

Reply:
`params->data = VERSION` will cause segmentation failures if libcrypto tries to read the values of those gettable_params using the utility getters.